### PR TITLE
For #22852 - Add a new "Cross-site cookies" permission

### PR DESCRIPTION
**Pull request by Mugurell**
Needs https://github.com/mozilla-mobile/android-components/pull/11399

Saw that once the permission is allowed GV would not request again this permission so atm changing the status from `Allowed` to `Blocked` will not change anything in the behavior. Opened https://bugzilla.mozilla.org/show_bug.cgi?id=1746031 for this situation.

https://user-images.githubusercontent.com/11428869/146388022-cb41bcba-c760-422e-b612-70aaae2784f4.mp4

cc @topotropic 

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/layout/quicksettings_permissions.xml [hunk 1]
```diff
@@ -136,12 +136,38 @@
...
         app:layout_constraintStart_toEndOf="@id/persistentStorageLabel"
         tools:text="@string/preference_option_phone_feature_blocked"
         tools:visibility="visible" />
 
+    <TextView
+        android:id="@+id/storageAccessLabel"
+        style="@style/QuickSettingsText.Icon"
+        android:layout_width="0dp"
+        android:layout_height="@dimen/quicksettings_item_height"
+        app:drawableStartCompat="@drawable/ic_storage"
```
Comment (Amejia481) [inline app/src/main/res/layout/quicksettings_permissions.xml:150]: Nit: Maybe we could use a [cookie icon](https://design.firefox.com/icons/viewer/#cookie)?  :)

**Review by Amejia481 — APPROVED** [review]
So far so good!
Just one nit https://github.com/mozilla-mobile/fenix/pull/22855/files#r769040253

**Comment by Mugurell** [comment]
Waiting for https://github.com/mozilla-mobile/android-components/pull/11399 to be merged.

**Comment by Mugurell** [comment]
Changed the permission name to "Cross-site cookies" - the same as it is on desktop.

**Pull request merged**